pkg/packet/bgp/bgp.go

```diff
@@ -12325,6 +12360,49 @@ func NewPathAttributeUnknown(flags BGPAttrFlag, typ BGPAttrType, value []byte) *
 	}
 }
 
+// BGPUpdateAttributes defines a map with a key as bgp attribute type
+// and value as bool. Value set to true indicates that the attribute specified by the key
+// exists in the bgp update.
+type BGPUpdateAttributes struct {
+	Attribute map[BGPAttrType]bool
+}
+
+func GetBGPUpdateAttributes(data []byte) map[BGPAttrType]bool {
+	m := make(map[BGPAttrType]bool)
+	for p := 0; p < len(data); {
+		flag := data[p]
+		p++
+		t := data[p]
```

Comment:
this line may panic on an incomplete input: better check if p can still be a valid index.

Reply:
and the same applies to the code below.

Reply:
that is going to be a lot of checks :)  I would let the repo owner comment since checking every increment of a pointer is not a common practice in this repo. I agree it is safer though..